From: James Rowe <jnrowe@gmail.com>
Subject: Use user flags
Status: LOCAL

Upstream appears dead, so a quick hack and not an upstreamable patch.

diff --git i/Makefile w/Makefile
index 0af3a63f35f3..47a709290936 100644
--- i/Makefile
+++ w/Makefile
@@ -7,11 +7,13 @@ LIBNAME = inotify.so
 # out this line when debugging.
 OMIT_FRAME_POINTER = -fomit-frame-pointer
 
+PKG_CONFIG ?= pkg-config
+
 # Seach for lua .pc file
-LUAPKG_CMD = $(shell pkg-config --list-all | grep Lua | awk '{print $$1}')
-CFLAGS = -fPIC -O3 -Wall $(shell pkg-config "$(LUAPKG_CMD)" --cflags)
-LFLAGS = -shared $(OMIT_FRAME_POINTER)
-INSTALL_PATH = $(shell pkg-config "$(LUAPKG_CMD)" --variable=INSTALL_CMOD)
+LUAPKG = lua
+CFLAGS += -fPIC -Wall $(shell $(PKG_CONFIG) $(LUAPKG) --cflags)
+LDFLAGS += -shared
+INSTALL_PATH = $(shell $(PKG_CONFIG) $(LUAPKG) --variable=INSTALL_CMOD)
 
 ## If your system doesn't have pkg-config, comment out the previous
 ## lines and uncomment and change the following ones according to your
@@ -27,10 +29,10 @@ $(OBJNAME): $(SRCNAME)
 	$(CC) -o $(OBJNAME) -c $(SRCNAME) $(CFLAGS)
 
 $(LIBNAME): $(OBJNAME)
-	$(CC) -o $(LIBNAME) -shared $(OBJNAME) $(LFLAGS)
+	$(CC) -o $(LIBNAME) -shared $(OBJNAME) $(LDFLAGS)
 
 install: $(LIBNAME)
-	install -D -s $(LIBNAME) $(DESTDIR)$(INSTALL_PATH)/$(LIBNAME)
+	install -D $(LIBNAME) $(DESTDIR)$(INSTALL_PATH)/$(LIBNAME)
 
 clean:
 	$(RM) $(LIBNAME) $(OBJNAME)
